### DISCUSSION TITLE
Hello Iam Newbie! I need help for my arduino

### DISCUSSION THREAD

**Discussion by NoobProgrammer32** [community/community#21445] [Other Feature Feedback, Questions, & Ideas]
<p>I just start study prigramming, and i am korean. I am hard to english, Sorry.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/9/49fc89d36d031b85a814d4e4fe6c94915f4a32f4.jpeg" data-download-href="/uploads/short-url/ayvUOntXxIMQu0fTTd9VgW7P5QM.jpeg?dl=1" title="20210731_234430" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/88241308/181064582-fbc487c8-0fb3-437c-a005-b376bbe46ff9.jpeg" alt="20210731_234430" data-base62-sha1="ayvUOntXxIMQu0fTTd9VgW7P5QM" width="666" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/9/49fc89d36d031b85a814d4e4fe6c94915f4a32f4_2_666x500.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/4/9/49fc89d36d031b85a814d4e4fe6c94915f4a32f4_2_999x750.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/4/9/49fc89d36d031b85a814d4e4fe6c94915f4a32f4_2_1332x1000.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/9/49fc89d36d031b85a814d4e4fe6c94915f4a32f4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">20210731_234430</span><span class="informations">1920×1440 118 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I was write code in arduino:</p>
<p>int d;</p>
<p>void setup() {<br>
pinMode(2, INPUT_PULLUP);<br>
pinMode(4, INPUT_PULLUP);<br>
pinMode(7, OUTPUT);<br>
Serial.begin(9600);<br>
d = 0;<br>
}<br>
void loop() {<br>
if(digitalRead(4) == LOW)<br>
d = 1;</p>
<pre><code>   if(digitalRead(2) == LOW)
   d = 0;

   if(d == 1)
   digitalWrite(7, HIGH);

   if(d == 0)
   digitalWrite(7, LOW);
   
   Serial.println(d);
</code></pre>
<p>}</p>
<p>And i test it<br>
But i was surprised<br>
Because i push off button(2), … Led is bright…?<br>
And push off, led is off again, i want hold turn on the light.</p>
<p>So i push on button(4), … Nothing…<br>
Pls, i am newbie and korean<br>
Dont hate me :sob:</p>

**Comment by tajmone** [comment]
<p>Hi <a class="mention" href="/noobprogrammer32">@NoobProgrammer32</a>, and welcome to the community!</p>
<p>Although I’m sure you’ll find many Arduino programmers here, my advise would be to look for support in an Arduino-specific community, where your question has less risk of going by unnoticed, and where the chances of finding people with the right expertise to answer you are higher — the Arduino experts you’ll find here are most likely also members of Arduino-specific forums anyway.</p>
<aside class="quote no-group" data-username="NoobProgrammer32" data-post="1" data-topic="193162">
<div class="title">
<div class="quote-controls"></div>
 NoobProgrammer32:</div>
<blockquote>
<p>and i am korean. I am hard to english, Sorry.</p>
</blockquote>
</aside>
<p>nothing to be sorry about. I’m sure you’ll find other Korean speaking users here. You can always avail yourself of the help of online translators:</p>
<p><a href="https://www.translate.com/korean-english" class="onebox" target="_blank" rel="noopener nofollow ugc">https://www.translate.com/korean-english</a></p>
<p>As for the Arduino question, I’m afraid I can’t help you with that.</p>

**Comment by NoobProgrammer32** [comment]
<p>Oh, github is not support arduino… Thanks!</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="NoobProgrammer32" data-post="3" data-topic="193162">
<div class="title">
<div class="quote-controls"></div>
 NoobProgrammer32:</div>
<blockquote>
<p>Oh, github is not support arduino…</p>
</blockquote>
</aside>
<p>I didn’t say that. Don’t get me wrong. I’m just saying that at the official Arduino community you’re likely to find better and quicker support:</p>
<ul>
<li><a href="https://forum.arduino.cc/" rel="noopener nofollow ugc">https://forum.arduino.cc/</a></li>
<li><a href="https://discord.com/invite/jQJFwW7" rel="noopener nofollow ugc">Arduino at Discord</a></li>
</ul>

**Comment by usuargithub** [comment, marked as answer]
<p>Hy,<br>
why did you draw the connexions that complicate?<br>
For me it seams to be ok, but it can be drawn much simpler.<br>
And with the program I also can’t help much, but it seams, that you ask for low, not for high. If butom is open, input is low, so you have to ask for high, not for low to detect pushing.<br>
Did you finde help in the links?</p>
<p>Try this:<br>
…</p>
<p>if(digitalRead(4) == HIGH)<br>
d = 1;</p>
<p>if(digitalRead(2) == HIGH)<br>
d = 0;</p>
<pre><code class="lang-auto"></code></pre>

**Comment by NoobProgrammer32** [comment]
<p>Oh Thank you very much!</p>

**Comment by usuargithub** [comment]
<p>Hello,<br>
did it work?</p>
<p>Regards</p>
